api/v1beta1/azuremachine_types.go

```diff
@@ -149,6 +152,24 @@ type SpotVMOptions struct {
 	EvictionPolicy *SpotEvictionPolicy `json:"evictionPolicy,omitempty"`
 }
 
+// SystemAssignedIdentityRole defines the role and scope to assign to the system assigned identity.
+type SystemAssignedIdentityRole struct {
+	// Name is the name of the role assignment to create for a system assigned identity. It can be any valid GUID.
```

Comment:
Should we say `UUID` rather than `GUID` here and on the next line? The two are really the same, but `GUID` is bit more Microsoft-specific, and the code that populates it calls `newUUID()`.

Reply:
UUID is definitely the correct term here